includes/wikia/parser/templatetypes/TemplateTypesParser.class.php

```diff
@@ -35,6 +36,71 @@ public static function onFetchTemplateAndTitle( &$text, &$finalTitle ) {
 	}
 
 	/**
+	 * @desc change template wikitext according to template type
+	 *
+	 * @param $templateTitle
+	 * @param $templateWikitext
+	 * @return bool
+	 */
+	public static function onBraceSubstitution( $templateTitle, &$templateWikitext ) {
+		global $wgEnableTemplateTypesParsing, $wgArticleAsJson, $wgCityId;
+
+		wfProfileIn( __METHOD__ );
+
+		if ( $wgEnableTemplateTypesParsing
+			&& $wgArticleAsJson
+			&& !empty( $templateWikitext )
+			&& $title = self::getValidTitle( $templateTitle ) ) {
+			$type = ( new ExternalTemplateTypesProvider( new \TemplateClassificationService ) )
+				->getTemplateTypeFromTitle( $wgCityId, $title );
+
+			if ( $type == AutomaticTemplateTypes::TEMPLATE_CONTEXT_LINK ) {
+				$templateWikitext = self::handleContextLinksTemplate( $templateWikitext );
+			}
+		}
+
+		wfProfileOut( __METHOD__ );
+
+		return true;
+	}
+
+	/**
+	 * @desc sanitize context-link template content - remove all non-link and non-text
+	 * elements from context-link template output and wrap it in div with special class
```

Comment:
Having 'and' in the description is a nice indicator that you may want to have sanitize() and wrap() methods called inside.